terraform/modules/rds/main.tf

```diff
@@ -0,0 +1,56 @@
+# Non-compliant RDS configuration (missing encryption, backups, and monitoring)
+resource "aws_db_instance" "main" {
+  identifier        = "${var.environment}-db"
+  engine           = "postgres"
+  engine_version   = "13.7"
+  instance_class   = var.instance_class
+  allocated_storage = 20
+
+  # Non-compliant: Credentials in plain text
+  name     = var.db_name
+  username = "admin"  # Hardcoded credentials
+  password = "insecure123!"  # Hardcoded password
```

Comment:
The RDS instance configuration is non-compliant with several SOC2 controls. Ensure that the RDS instance has storage encryption enabled by setting `storage_encrypted = true`. This is critical for protecting data at rest.